.github/workflows/build-and-push.yml

```diff
@@ -0,0 +1,38 @@
+on:
+    push:
+        branches:
+            - master
+    pull_request:
+        branches:
+            - "*"
+
+name: Build and Push Image
+
+jobs:
+    test:
+        runs-on: ubuntu-latest
+        steps:
+            - uses: actions/checkout@v4
+
+            - name: Login to Docker Hub
+              uses: docker/login-action@v3
+              with:
+                username: ${{ secrets.DOCKERHUB_USERNAME }}
+                password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+            - name: Make shared env
+              uses: ./.github/actions/make-shared-env
+
+            - name: Build app
+              env:
+                NODE_ENV: production
+              run: src/gradlew -p src :app:bootDistTar
```

Comment:
Why d we need to build the app here?

Reply:
the dockerfile just copies the build files, we dont build the app at any point before this